NuRadioReco/modules/trigger/rnog_surface_trigger.py

```diff
@@ -0,0 +1,214 @@
+import copy
+import time
+import logging
+
+logger = logging.getLogger('rnog_surface_trigger')
+
+def schottky_diode(trace, threshold, temp=250*units.kelvin, Vbias=2*units.volt):
+    '''
+    Returns the absolute maximum of the diode response depending on the maximum input signal.
+    Options for temperature are [300K, 273K, 250K], for the bias voltages [2, 1.5, 1, 0.5]V.
+    The triggerpath reduces the voltage by -10dB, hence the threshold is small. The linear fit is obtained from
+    measurements done at DESY Zeuthen.
+
+    Parameters
+    ----------
+    trace: array of floats
+        the signal trace
+    threshold: float
+        the threshold
+    temp: float
+        temperature of the board with diode
+    Vbias: float
+        applied bias voltage to the diode board
+        
+    Returns
+    -------
+    trigger: list of booleans
+        trigger evaluation for each entry of the trace
+    '''
+
+    def func(x, a, b):
+        return a * x + b
+
+    if temp == 300 *units.kelvin:  #measurements taken at 300K
+        if Vbias == 2*units.volt:
+            a = 38.10032
+            b = -9.19654194e-08
+        if Vbias == 1.5*units.volt: #measurements taken at 1.6V
+            a = 34.6274877
+            b = -8.45140238e-05
+        if Vbias == 1*units.volt:
+            a = 24.6683322
+            b = -1.03679002e-03
+        if Vbias == 0.5*units.volt:  #measurements taken at 0.6V
+            a = 16.0005295
+            b = -5.83972425e-04
+
+    if temp == 273*units.kelvin: #measurements taken at 273K
+        if Vbias == 2*units.volt:
+            a = 45.9684369
+            b = -6.15792586e-06
+        if Vbias == 1.5*units.volt:
+            a = 38.80661
+            b = -3.01491791e-04
+        if Vbias == 1*units.volt:
+            a = 29.45406
+            b = -7.71227505e-04
+        if Vbias == 0.5*units.volt:
+            a = 15.4192195
+            b = -5.10530795e-04
+
+    if temp == 250*units.kelvin: #measurements taken at 248K
+        if Vbias == 2*units.volt:
+            a = 55.0380132
+            b = -2.13447979e-05
+        if Vbias == 1.5*units.volt:
+            a = 49.3192486
+            b = -1.67121000e-03
+        if Vbias == 1*units.volt:
+            a = 37.4351427
+            b = -9.76368002e-04
+        if Vbias == 0.5*units.volt:
+            a = 17.4574667
+            b = -5.90216182e-04
+
+    v_in = (trace)**2
+    v_out = func(v_in, a, b)
+    return v_out > threshold
+
+class triggerSimulator:
+    """
+    Calculates the RNO_G surface trigger with a bandpass filter, a -10db attenuator and a schottky_diode.
+    """
+
+    def __init__(self):
+        self.__t = 0
+        self.begin()
+
+    def begin(self):
+        return
+
+    @register_run()
+    def run(self, evt, station, det, threshold, coinc_window=60*units.ns, number_coincidences=1, triggered_channels=[13, 16, 19], temp=250*units.kelvin, Vbias=2*units.volt, trigger_name='rnog_surface_trigger'):
+        """
```

Comment:
docstring nitpickier nitpicking - methods look nicer in sphinx if they have at least a single line description. Would be nice if you could add one for these methods too (can be trivial, e.g. `run the trigger module`).

Reply:
Thanks, I will do that